### ISSUE TITLE
Better Questing 'random crash'

### ISSUE THREAD

**Issue by DarkTechnician** [CaiganMythFang/TheFerretBusinessIssues#18]
So.. I did the switch/revert to the previous version of BQ (And Standard Expansion) as is suggested in the changelog for the new version, and crashed a LOT still. I switched to the original pack version, and still had just as many crashes. Then I noticed what appeared to be happening at the same time the crashing was happening. (hint: same time) The crash was being caused by a Concurrent Modification error. (Which we most likely also already knew) Well, I have good news. It isn't Better Questing causing the issue, even though that's what was being blamed in the stack. It's Inventory Tweaks. I ALWAYS was happening to be sorting my (player) inventory at the same time I was crashing. I turned off the hotkey to sort via hotkey (along with the auto-refill option) and haven't had a Concurrent Modification crash since. (Appears that BetterQuesting was trying to read the player's inventory while Inventory Tweaks was trying to SORT the same inventory, aka a Concurrent Modification.)

Tl;Dr: I found a workaround/fix. Disable/turn off Sorting of player inventory in Inventory Tweaks.

**Comment by CaiganMythFang** [comment]
This might be one cause of the issue, or a separate crash cause with a similar stack trace. I've had it occur to myself randomly, and I don't use Inventory Tweaks auto sorting OR hotkey sorting (only time it will sort is if I hit the button in a container)

Could you get me a crash log from it happening with the inventory sorting method? I'm trying to reproduce it on my end and am unable to.

**Comment by DarkTechnician** [comment]
Yep, just happened before I checked on this, in fact..
previous 10 mins (before crash, including log report) of full log:
http://pastebin.com/31KKPuSC
Crashlog:
http://pastebin.com/G4GMKRRr

**Comment by CaiganMythFang** [comment]
Alright, going to keep digging into this, thank you.

**Comment by DarkTechnician** [comment]
No Problem. I'm pretty sure it wouldn't be too hard to talk to Better Questing and Inventory Tweaks, and get them to fix their Concurrent Modification problems.. (I seem to remember watching a video once about how to fix those.. might only be available in java 7 or 8 though)

**Comment by DarkTechnician** [comment]
By the way, don't revert the pack to the previous BQ. I just had the instance happen where it fails/doesn't load the quests properly. (Hopefully it didn't save them while they were messed up..) Seems to make them slowly become 'New Quest' with no objectives, and automatically complete.

**Comment by Ristellise** [comment]
BQ1 is a mess.... just sayin...

**Comment by CaiganMythFang** [comment]
0.2.2 Update is out, should fix the HashMap issue. For existing worlds, please be sure to do the Quest Tablet Firmware Update in the Initial Research quest page, and then the new quest connected to it ([ADMIN] Scoreboard and Progress Reset)
